input-output-hk/cardano-sl
# Akegalj/co 319/swagger account index

## Description

The example accountId from the API doc isn't much relevant and reflects an invalid value

## Linked issue

https://iohk.myjetbrains.com/youtrack/issue/CO-319

## Type of change


## Developer checklist


## Testing checklist


## QA Steps
1) Run a cardano-node
2) Visit https://localhost:8091/docs/v1/index/ - you will see changes showed bellow
3) Roundtrip tests pass (the screenshot bellow)

## Screenshots (if available)
`accountIndex` in request body shows correct value range
![look](https://user-images.githubusercontent.com/2461951/41376755-76dc7a74-6f5a-11e8-9e0c-8e2bf54b7cff.png)


`accountIndex` on description on the right picks correct value (not bellow the minimum)
![look](https://user-images.githubusercontent.com/2461951/41376784-88421a08-6f5a-11e8-98c6-b324257eace6.png)


`accountId` (same thing as `accountIndex`) in request params shows correct value range
![look](https://user-images.githubusercontent.com/2461951/41376464-96e8cd6e-6f59-11e8-8a3e-f967d41ac076.png)

roundtrip tests
![look](https://user-images.githubusercontent.com/2461951/41376705-4a2977d4-6f5a-11e8-936d-309a09b92358.png)


@KtorZ I wonder why did we ended up with `accountId` and `accountIndex` json keys for the same thing. Is it still too late to unify it? (it would be a breaking change for exchanges)
